### ISSUE TITLE
Not able to receive media messages all of a sudden.

### ISSUE THREAD

**Issue by sgowris2** [signalapp/Signal-Android#7992]
- [ X] I have searched open and closed issues for duplicates
- [X ] I am submitting a bug report for existing functionality that does not work as intended
- [ X] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [ X] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Any media messages that I am receiving are not auto-downloading even though the settings are set to auto-download. And when I click on download on the image, the loader spins forever. It started happening with one particular image and ever since the issue has not gone away. I have deleted that chat, reset secure session, restarted phone, checked my data connection. I am able to send pictures and they work just fine.

### Steps to reproduce
Not sure what caused this or how to reproduce.

### Device info

**Device:** OnePlus 3
**Android version:** 8.0.0
**Signal version:** 4.23.4

### Link to debug log
https://debuglogs.org/cb53c8664ab9c77abb4d9c20f82ebe1d9bfa21301252f63f5d1c0ce82adb762b

**Comment by greyson-signal** [comment]
- Is this happening for all future incoming media messages, or is just one message stuck?
- Is the person you're talking with added to your contacts?
- Is this a group chat?
- Is this a chat with yourself?

**Comment by sgowris2** [comment]
1. This is happening for all future incoming messages.
2. It's happening with all chats. They are all in my contacts.
3. Have not tried it in a group chat. 
4. This is not a chat with myself.

Update: I uninstalled and re-installed the app, and everything is working again. Unfortunately, I can't go back and get more logs for the issue unless it happens again.

**Comment by greyson-signal** [comment]
Ah shoot, yeah, I was gonna ask you to get more logs lol. I'll do my best to repro, but that tends to be tricky with these rare situations.

**Comment by Spork-Schivago** [comment]
If it helps any, my friend tried sending me a video through Signal.   We both have it, he is in my contacts, but the video would not play.   I can try having other people with signal try to send me a video.   Perhaps this is related?

I can send pictures successfully though.   I have not tried to receive any.

**Comment by Wasseranomalie** [comment]
This happened on my phone too, after receiving an image in a group while loosing the WLAN-signal several times during the download. Now this image and a video I received later won't download anymore. All contacts in this group are in my contact list.

https://debuglogs.org/ca0b433006d9cf29a5dcfce26f1689abdb6cd7d4192e62fc82b11d51f917b4ea

**Comment by duereg** [comment]
https://debuglogs.org/a31345d175e34f6c364253b43ebf90c73774a492567504bd4fd6c03fb53d53cc

Similar issue happening to me. I can't seem to download any of my MMS messages all of a sudden.

**Comment by RyanMarcus** [comment]
Same issue. 

Happens in both individual ans group chats. Does *not* occur in a chat with myself. 


https://debuglogs.org/6838785d25da454384a487bbc409c322a9133bb127a6a781a6fa49af97dd283e

**Comment by TheRealCuran** [comment]
I've got this problem as well. The first time happened with a shaky network. The desktop client continues to download the media.

[NB: MMS download never worked for me, but since I know only one person who sent me MMS but now has Signal as well, I do not really care about that part.]

**Comment by greyson-signal** [comment]
@TheRealCuran Can you post a debug log?

**Comment by TheRealCuran** [comment]
Sure: https://debuglogs.org/5e2db280111ac04ef7ccd2e70bdea2b40d4601ed1cbc2d085c19a1bef6545cc8

Though the "memory free" line looks suspicious. Android tells me, that I'm using a bit over 50 % of the 2 GB installed.

**Comment by greyson-signal** [comment]
@TheRealCuran You're hitting an issue that is fixed in 4.31.x, which is rolling out to production now.
@RyanMarcus  You had the same problem.

```java
2018-12-08 22:12:35.496 CET I Job: [41759ce6-4d32-4b37-8d65-43981f8698fc] MultiDeviceContactUpdateJob :: Retrying due to unmet requirements. (Time since submission: 186267701 ms, Run attempt: 82, isStopped: false)
```

Thanks!

**Comment by TheRealCuran** [comment]
I can confirm that the 4.31.x branch fixes the issue for me, just got 4.31.5 and media files are loading again. Thank you.
